locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -204,7 +205,8 @@
     private static final Pattern LIST_PATTERN = Pattern.compile("^\\[(.*)\\]$");
     private static final Pattern INTEGER_PATTERN = Pattern.compile("^\\d*$");
 
-    private static boolean loggedSshKeysHint = false;
+    private final AtomicBoolean loggedSshKeysHint = new AtomicBoolean(false);
+    private final AtomicBoolean listedAvailableTemplatesOnNoSuchTemplate = new AtomicBoolean(false);
```

Comment:
a guava `Cache` is maybe the hyper-engineered way but i think tying it to the `JcloudsLocation` instance is a pragmatic step.  the `JcloudsLocation` instance should be cached, i *think*, and if not all we do is log some debug/info a bit more.